batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/internal/core/impl/BatchProject.java

```diff
@@ -0,0 +1,632 @@
+		if(jp == null) return null;
+		try {
+			return EclipseJavaUtil.findType(jp, qualifiedName);
+		} catch (JavaModelException e) {
+			BatchCorePlugin.pluginLog().logError(e);
+		}
+		return null;
+	}
+
+	public ParametedTypeFactory getTypeFactory() {
+		return typeFactory;
+	}
+
+	public ClassPathMonitor getClassPath() {
+		return classPath;
+	}
+
+	public void cleanTypeFactory() {
+		typeFactory.clean();
+		BatchProject[] ps = getAllDependentProjects();
+		for (BatchProject n: ps) {
+			n.typeFactory.clean();
+		}
+	}
+	
+	@Override
+	public Collection<IBatchArtifact> getAllArtifacts() {
+		Set<IBatchArtifact> result = new HashSet<IBatchArtifact>();
+		synchronized(this) {
+			result.addAll(allArtifacts);
+		}
+		return result;
+	}
+
+	@Override
+	public synchronized Collection<IBatchArtifact> getArtifacts(IResource resource) {
+		Set<IBatchArtifact> result = artifactsByPath.get(resource.getFullPath());
+		return result != null ? new HashSet<IBatchArtifact>(result) : null;
+	}
+
+	@Override
+	public synchronized IBatchArtifact getArtifact(IType type) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public synchronized Set<IFile> getDeclaredBatchJobs() {
+		Set<IFile> result = new HashSet<IFile>();
+		for (BatchJobDefinition def: getDefinitions().getBatchJobDefinitions()) {
+			result.add(def.getFile());
+		}
+		return result;
+	}
+
+	@Override
+	public synchronized Collection<IBatchArtifact> getArtifacts(String name) {
+		Set<IBatchArtifact> result = artifactsByName.get(name);
+		return result != null ? new HashSet<IBatchArtifact>(result) : EMPTY;
+	}
+
+	static Collection<IBatchArtifact> EMPTY = Collections.EMPTY_SET;
+
+	@Override
+	public synchronized Collection<IBatchArtifact> getArtifacts(BatchArtifactType type) {
+		Collection<IBatchArtifact> result = artifactsByType.get(type);
+		return result != null ? new HashSet<IBatchArtifact>(result) : EMPTY;
+	}
+	/**
+	 * Returns true, if model is fully loaded.
+	 * 
+	 * @return
+	 */
+	@Override
+	public boolean isStorageResolved() {
+		return isStorageResolved;
+	}
+
+	public void store() throws IOException {
+		isBuilt = true;
+//		File file = getStorageFile();
+	}
+
+	public boolean hasNoStorage() {
+		if(isBuilt) return false;
+		File f = null;// getStorageFile();
+		return f == null || !f.exists();
+	}
+
+	public void resolveStorage(boolean load) {
+		if(isStorageResolved) return;
+		if(load) {
+			load();
+		} else {
+			isStorageResolved = true;
+		}
+	}
+
+	public void resolve() {
+		resolveStorage(true);
+	}
+
+	public void load() {
+		if(isStorageResolved) return;
+		isStorageResolved = true;
+		try {
+			new BatchBuilder(this);
+		} catch (CoreException e) {
+			BatchCorePlugin.pluginLog().logError(e);
+		}
+		
+//		postponeFiring();
+		
+		try {		
+			//Use kb storage for dependent projects since cdi is not stored.
+			loadProjectDependenciesFromKBProject();
+		} finally {
+//			fireChanges();
+		}
+	}
```

Comment:
Let's clean up the code. Remove unused try,final. "cdi"->"Batch" in the comment. Unused method invocations.